src/components/LineItemCategory.js

```diff
@@ -1,30 +1,79 @@
 import React from 'react';
 import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
 import PropTypes from 'prop-types';
-import { Feather } from '@expo/vector-icons';
+import {
+  Feather,
+  Entypo,
+  MaterialIcons,
+  MaterialCommunityIcons,
+  FontAwesome
+} from '@expo/vector-icons';
 import { colors, fonts, gStyle } from '../constants';
 
-const LineItemCategory = ({ icon, onPress, title }) => (
-  <TouchableOpacity
-    activeOpacity={gStyle.activeOpacity}
-    onPress={onPress}
-    style={styles.container}
-  >
-    <View style={gStyle.flexRowCenterAlign}>
-      <Feather color={colors.greyInactive} name={icon} size={24} />
-      <Text style={styles.title}>{title}</Text>
-    </View>
-    <View style={styles.containerRight}>
-      <Feather color={colors.greyInactive} name="chevron-right" size={20} />
-    </View>
-  </TouchableOpacity>
-);
+const LineItemCategory = ({
+  icon,
+  onPress,
+  title,
+  disableRightSide,
+  iconLibrary
+}) => {
+  // TODO: spread props
+  const Icon = {
+    Feather: <Feather color={colors.greyInactive} name={icon} size={24} />,
+    Entypo: <Entypo color={colors.greyInactive} name={icon} size={24} />,
+    MaterialCommunityIcons: (
+      <MaterialCommunityIcons
+        color={colors.greyInactive}
+        name={icon}
+        size={24}
+      />
+    ),
+    MaterialIcons: (
+      <MaterialIcons color={colors.greyInactive} name={icon} size={24} />
+    ),
+    FontAwesome: (
+      <FontAwesome color={colors.greyInactive} name={icon} size={24} />
+    )
+  };
```

Comment:
this is the biggest pain point, it's rendering each lib, even when it's not being used, and causing a lot of warning message for each iteration, because some of the icon names are not supported in that library, so try to break this out a bit.

Reply:
this also breaks the library screen with a lot of warnings for the same reason, maybe refactor to just accept an icon element?

Reply:
okay, to prevent all of the unnecessary renderings of other vector icon libraries, try to use a switch statement:
```
let iconDisplay;

  switch (iconLibrary) {
    default:
    case 'Feather':
      iconDisplay = (
        <Feather color={colors.greyInactive} name={icon} size={24} />
      );
      break;
    case 'Entypo':
      iconDisplay = (
        <Entypo color={colors.greyInactive} name={icon} size={24} />
      );
      break;
    case 'MaterialIcons':
      iconDisplay = (
        <MaterialIcons color={colors.greyInactive} name={icon} size={24} />
      );
      break;
    case 'MaterialCommunityIcons':
      iconDisplay = (
        <MaterialCommunityIcons
          color={colors.greyInactive}
          name={icon}
          size={24}
        />
      );
      break;
    case 'FontAwesome':
      iconDisplay = (
        <FontAwesome color={colors.greyInactive} name={icon} size={24} />
      );
      break;
  }
```

this will remove/replace the entire Icon object. the just have ```{iconDisplay}``` instead of 
```{iconLibrary ? (
          Icon[iconLibrary]
        ) : (
          <Feather color={colors.greyInactive} name={icon} size={24} />
        )}
```
also update the props validation to be:
```
LineItemCategory.defaultProps = {
  ...
  disableRightSide: null,
  iconLibrary: 'Feather'
};

LineItemCategory.propTypes = {
  ...
  disableRightSide: PropTypes.bool
};